packages/cactus-plugin-persistence-fabric/src/test/typescript/integration/persistence-fabric-db-client.test.ts

```diff
@@ -0,0 +1,233 @@
+  const testPluginInstanceId = "testInstance";
+  let postgresContainer: PostgresTestContainer;
+  let dbClient: PostgresDatabaseClient;
+
+  //////////////////////////////////
+  // Environment Setup
+  //////////////////////////////////
+
+  beforeAll(async () => {
+    log.info("Prune Docker...");
+    await pruneDockerAllIfGithubAction({ logLevel: testLogLevel });
+
+    log.info("Run PostgresTestContainer...");
+    postgresContainer = new PostgresTestContainer({
+      imageName: postgresImageName,
+      imageVersion: postgresImageVersion,
+      logLevel: testLogLevel,
+      envVars: ["POSTGRES_USER=postgres", "POSTGRES_PASSWORD=postgres"],
+    });
+    await postgresContainer.start();
+
+    const postgresPort = await postgresContainer.getPostgresPort();
+    expect(postgresPort).toBeTruthy();
+    log.info(`Postgres running at localhost:${postgresPort}`);
+
+    log.info("Create PostgresDatabaseClient");
+    dbClient = new PostgresDatabaseClient({
+      connectionString: `postgresql://postgres:postgres@localhost:${postgresPort}/postgres`,
+      logLevel: sutLogLevel,
+    });
+
+    log.info("Connect the PostgreSQL PostgresDatabaseClient");
+    await dbClient.connect();
+    await dbClient.client.query(
+      `CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+       CREATE ROLE anon NOLOGIN;
+       CREATE ROLE authenticated NOLOGIN;
+       CREATE ROLE service_role NOLOGIN;
+       CREATE ROLE supabase_admin NOLOGIN;`,
+    );
+
+    log.info("Initialize the test DB Schema");
+    await dbClient.initializePlugin(testPluginName, testPluginInstanceId);
+
+    log.info("Mock Supabase schema");
+    log.info("Ensure DB Schema is empty (in case test is re-run on same DB");
+  }, setupTimeout);
+
+  afterAll(async () => {
+    log.info("FINISHING THE TESTS");
+
+    if (dbClient) {
+      log.info("Disconnect the PostgresDatabaseClient");
+      await dbClient.shutdown();
+    }
+
+    if (postgresContainer) {
+      log.info("Stop PostgreSQL...");
+      await postgresContainer.stop();
+    }
+    log.info("Prune Docker...");
+    await pruneDockerAllIfGithubAction({ logLevel: testLogLevel });
+  }, setupTimeout);
+
+  //insert simple BlockData into db
+  test("insert into fabric_blocks", async () => {
+    const block_data = {
+      fabric_block_id: "1",
+      fabric_block_num: 1,
+      fabric_block_data:
+        "dd9c20b516743ac44da1afe5242b38baa4c7523bb66f8473ae94299f1fbff810",
+    };
+    const response = await dbClient.insertBlockDataEntry(block_data);
+
+    expect(response).toBeTruthy();
+    expect(response.command).toEqual("INSERT");
+    expect(response.rowCount).toEqual(1);
+    expect(response).toMatchObject({
+      command: "INSERT",
+      rowCount: expect.toBeNumber(),
+      oid: expect.toBeNumber(),
+      rows: expect.toBeArray(),
+      fields: expect.toBeArray(),
+      _types: expect.toBeObject(),
+      RowCtor: null,
+      rowAsArray: expect.toBeBoolean(),
+    });
+  });
+  // check simple database query
```

Comment:
Can you insert a blank line above the comment?

Reply:
> Can you insert a blank line above the comment (better readability)?

done